addons/account/models/account_invoice.py

```diff
@@ -1045,6 +1045,7 @@ def refund(self, date_invoice=None, date=None, description=None, journal_id=None
                 'in_invoice': ('vendor bill refund')}
             message = _("This %s has been created from: <a href=# data-oe-model=account.invoice data-oe-id=%d>%s</a>") % (invoice_type[invoice.type], invoice.id, invoice.number)
             refund_invoice.message_post(body=message)
+            invoice.refund_invoice_id = refund_invoice
```

Comment:
is it normal to change behavior of refund_invoice_id field ? I think, it will be created problem in migration. 

and any reason to check refund_invoice_id in purchase module on creating new invoice ?

see code: https://github.com/odoo/odoo/blob/master/addons/purchase/models/account_invoice.py#L189